packages/renderer/src/lib/chat/components/chat.svelte

```diff
@@ -21,17 +24,31 @@ import NoModelsAvailable from './NoModelsAvailable.svelte';
 let {
   chat,
   readonly,
-  initialMessages,
+  messages,
 }: {
   chat: DbChat | undefined;
-  initialMessages: UIMessage[];
+  messages: DbMessage[];
   readonly: boolean;
 } = $props();
 
 let models: Array<ModelInfo> = $derived(getModels($providerInfos));
 
-let selectedModel = $derived<ModelInfo | undefined>(models[0]);
-let selectedMCP = $state<MCPRemoteServerInfo[]>([]);
+// Partial because existing messages in database have been set an empty config
+const config = messages[messages.length - 1]?.config as Partial<MessageConfig> | undefined;
```

Comment:
issue: we should not any casting (if not in tests), it breaks all typechecking detection

Reply:
I agree to limit casting as much as possible but in this case the database helpers stores JSON and I don't know what to do, maybe a zod validation?